frontend/src/views/Join.tsx

```diff
@@ -67,23 +103,25 @@ function JoinGamePage() {
                 event.preventDefault();
               }
               if (event.key === 'Enter' && validRoomId) {
-                setPageState(2);
+                checkRoom();
               }
             }}
           />
           <LargeInputButton
             onClick={() => {
-              setPageState(2);
+              checkRoom();
             }}
             value="Enter"
-            // Input is disabled if no nickname exists, has a space, or is too long.
-            disabled={!validRoomId}
+            // Input is disabled if loading or if no nickname exists, has a space, or is too long.
+            disabled={!validRoomId || loading}
           />
           { focusInput && !validRoomId ? (
             <Text>
               The room ID must have exactly six digits (numbers 0 through 9).
             </Text>
           ) : null}
+          { loading ? <Loading /> : null }
+          { error ? <ErrorMessage message={error} /> : null }
         </div>
       );
       break;
```

Comment:
Cannot comment on an unchanged line, so this must've been on a separate PR (maybe mine). However, I just noticed that the comment on line 134 is no longer applicable, and should be removed.